File: SKIBin_old/app/src/main/java/com/h2Invent/skibin/ChildListMainActivity.java
package com.h2Invent.skibin;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.LinearLayout;

import com.facebook.drawee.backends.pipeline.Fresco;
import com.google.android.material.tabs.TabLayout;

public class ChildListMainActivity extends AppCompatActivity implements FragmentChildListAfterList.OnItemSelectedListener, FragmentChildListCheckin.OnItemSelectedListener {
    private static final String LOG = ChildListMainActivity.class.getSimpleName();

    private TabLayout childList;
    private ViewPager mViewPager;
    private TabChildListAdapter mChildListAdapter;
    private LinearLayout childListContainer;
    private String checkinUrl;
    private String kinderlisteUrl;
    public static String userToken;
    private static final int CAMERA_PERMISSION_CODE = 100;
    private static final int PHONE_PERMISSION_CODE = 101;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_child_list_main);
        Fresco.initialize(this);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_main);
        setSupportActionBar(toolbar);
        childList = (TabLayout) findViewById(R.id.tabChildList);
        childListContainer = (LinearLayout) findViewById(R.id.childListContainer);

    }
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_activity_main, menu);
        return true;
    }

    @Override
    protected void onResume() {
        super.onResume();
        initApp();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_get_data:
                initApp();
                return true;
            case R.id.action_settings:
                Intent intent = new Intent(ChildListMainActivity.this, TabSettingsActivity.class);
                startActivity(intent);
                return true;
            default:
                Log.d(LOG, "Debug       - Press something.");
                return super.onOptionsItemSelected(item);
        }
    }

    public  void initApp() {

        SharedPreferences sharedPreferences = getSharedPreferences(MainActivity.SHARED_PREFS, MODE_PRIVATE);

            childListContainer.setVisibility(View.VISIBLE);
            kinderlisteUrl = sharedPreferences.getString(SaveSettings.USER_KINDERLISTE_URL, "");
            checkinUrl = sharedPreferences.getString(SaveSettings.USER_CHECKIN_URL, "");
            userToken = sharedPreferences.getString(SaveSettings.USER_TOKEN,"");
            setViewPager();
    }
    private void setViewPager() {
        mViewPager = (ViewPager) findViewById(R.id.pager);
        mChildListAdapter = new TabChildListAdapter(getSupportFragmentManager());
        mViewPager.setAdapter(mChildListAdapter);
        final FragmentChildListCheckin fragmentChildListCheckin = (FragmentChildListCheckin) mChildListAdapter.getItem(0);
        fragmentChildListCheckin.setUrl(checkinUrl);
        fragmentChildListCheckin.setUserToken(userToken);
        final FragmentChildListAfterList fragmentChildListAfterList= (FragmentChildListAfterList) mChildListAdapter.getItem(1);
        fragmentChildListAfterList.setUserToken(userToken);
        fragmentChildListAfterList.setUrl(kinderlisteUrl);
        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabChildList);
        tabLayout.setupWithViewPager(mViewPager);
    }
}

File: SKIBin_old/app/src/main/java/com/h2Invent/skibin/ChildListMainActivity.kt
package com.h2Invent.skibin

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.viewpager.widget.ViewPager
import com.facebook.drawee.backends.pipeline.Fresco
import com.google.android.material.tabs.TabLayout

class ChildListMainActivity : AppCompatActivity(),
    FragmentChildListAfterList.OnItemSelectedListener,
    FragmentChildListCheckin.OnItemSelectedListener {

    private lateinit var childListContainer: LinearLayout
    private lateinit var childListAdapter: TabChildListAdapter
    private lateinit var viewPager: ViewPager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Fresco.initialize(this)
        setContentView(R.layout.activity_child_list_main)
        setSupportActionBar(findViewById<Toolbar>(R.id.toolbar_main))

        childListContainer = findViewById(R.id.childListContainer)
        viewPager = findViewById(R.id.pager)
        childListAdapter = TabChildListAdapter(supportFragmentManager)
        viewPager.adapter = childListAdapter
        findViewById<TabLayout>(R.id.tabChildList).setupWithViewPager(viewPager)
    }

    override fun onResume() {
        super.onResume()
        initApp()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_activity_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean = when (item.itemId) {
        R.id.action_get_data -> {
            initApp()
            true
        }
        R.id.action_settings -> {
            startActivity(Intent(this, TabSettingsActivity::class.java))
            true
        }
        else -> super.onOptionsItemSelected(item)
    }

    fun initApp() {
        val sharedPreferences = getSharedPreferences(SaveSettings.SHARED_PREFS, MODE_PRIVATE)
        childListContainer.visibility = View.VISIBLE

        val childListUrl = sharedPreferences.getString(SaveSettings.USER_KINDERLISTE_URL, "").orEmpty()
        val checkinUrl = sharedPreferences.getString(SaveSettings.USER_CHECKIN_URL, "").orEmpty()
        val userToken = sharedPreferences.getString(SaveSettings.USER_TOKEN, "").orEmpty()

        childListAdapter.checkinFragment.setConfig(checkinUrl, userToken)
        childListAdapter.afterListFragment.setConfig(childListUrl, userToken)
    }

    companion object {
        const val EXTRA_USER_TOKEN = "extra_user_token"
    }
}
